app/lib/differential_expression_service.rb

```diff
@@ -308,4 +308,27 @@ def self.log_message(message)
     puts message
     Rails.logger.info message
   end
+
+  # Get token for client-side access to GCS bucket, depending on study privacy
+  # Context: https://github.com/broadinstitute/single_cell_portal_core/pull/1747
+  def self.get_read_access_token(study, user, renew=false)
```

Comment:
Blocker: while this particular issue is scoped to the differential expression results table, the _actual_ underlying issue is much broader than that.  It can affect any client-side fetch of a remote object, not just DE files.  As such, this method doesn't really belong in the `DifferentialExpressionService` class.

That being said, we don't really have an obvious place to put this code outside of a controller.  I could see it being placed in `app/lib/request_utils.rb`, or possibly a new "service" class (though that may be overkill but possibly not).

Also, see the previous comment re: kwargs and boolean arguments.